goodkey/good_key.go

```diff
@@ -36,7 +36,7 @@ var smallPrimeInts = []int64{
 // singleton defines the object of a Singleton pattern
 var (
 	smallPrimesSingleton sync.Once
-	smallPrimes          []*big.Int
+	smallPrimes          *big.Int
```

Comment:
Might be worthwhile updating the name of this variable to indicate it is the product, but not super important.

Reply:
Updated. :)